### PR TITLE
feat: OAuthUseCase improvements

### PR DESCRIPTION
**Pull request by KaterinaWire**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

 1. Change the keycloak redirectURI, we don't need to have it different for every app.
 2. Set `prefersEphemeralSession` to `true`.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/OAuthUseCase.swift [hunk 1]
```diff
@@ -42,7 +42,7 @@ public class OAuthUseCase: OAuthUseCaseInterface {
         logger.info("invoke authentication flow")
 
         guard let bundleID = Bundle.main.bundleIdentifier,
-              let redirectURI = URL(string: "\(bundleID):/oauth2redirect")
+              let redirectURI = URL(string: "wire://e2ei/oauth2redirect")
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/OAuthUseCase.swift]: did you test it on C builds? as wire is not a scheme, does that work?
you can change the WIRE_URL_SCHEME to check locally?

File: wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/OAuthUseCase.swift [hunk 2]
```diff
@@ -109,7 +109,10 @@ public class OAuthUseCase: OAuthUseCaseInterface {
 
     @MainActor
     private func execute(authorizationRequest: OIDAuthorizationRequest) async throws -> OAuthResponse {
-        guard let userAgent = OIDExternalUserAgentIOS(presenting: rootViewController) else {
+        guard let userAgent = OIDExternalUserAgentIOS(
+            presenting: rootViewController,
+            prefersEphemeralSession: true)
+        else {
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/OAuthUseCase.swift]: 💅 Somehow I would rather format this way, but it's up to you:
```suggestion
        guard let userAgent = OIDExternalUserAgentIOS(
            presenting: rootViewController,
            prefersEphemeralSession: true
        ) else {
```

File: wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/OAuthUseCase.swift (hunk 1 above)
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/OAuthUseCase.swift]: Can it happen that users have multiple wire apps installed and then another app is called? 🤓

**Review by markusfassbender — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/OAuthUseCase.swift (hunk 1 above)
Comment (KaterinaWire) [reply wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/OAuthUseCase.swift]: `ASWebAuthenticationSession ensures that only the calling app’s session receives the authentication callback, even when more than one app registers the same callback URL scheme.`

https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession

Comment (KaterinaWire) [reply wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/OAuthUseCase.swift]: yes, with `WIRE_URL_SCHEME = wire-c3-test`

**Review by netbe — APPROVED** [review]

**Pull request merged**